source/runner/DangerfileRunner.ts

```diff
@@ -7,20 +7,28 @@ import { DangerContext } from "../runner/Dangerfile"
 
 import { NodeVM, NodeVMOptions } from "vm2"
 
-let hasTypeScript = false
 let hasBabel = false
+let hasTypeScript = false
+let hasFlow = false
 
 declare const regeneratorRuntime: any
 
 // You know you're being a dangerous badass when you have this many linter disables. Deal with it.
-try {
-  require.resolve("typescript") // tslint:disable-line
-  hasTypeScript = true
-} catch (e) {} // tslint:disable-line
+
 try {
   require.resolve("babel-core") // tslint:disable-line
   require("babel-polyfill") // tslint:disable-line
   hasBabel = true
+
+  try {
+    require.resolve("babel-plugin-transform-typescript") // tslint:disable-line
```

Comment:
Yeh, I thought about this for a bit, and came to the same conclusion. So I started work on bringing back the original typescript optional dep also - so you can have TypeScript via `tsc`, or Babel, and then optionally babel for JS. All depending on if the deps exist.